frontend/packages/console-app/src/components/network-policies/network-policy-rule-config.tsx

```diff
@@ -20,7 +21,11 @@ import { NetworkPolicyPeerIPBlock } from './network-policy-peer-ipblock';
 import { NetworkPolicyPeerSelectors } from './network-policy-peer-selectors';
 import { NetworkPolicyPorts } from './network-policy-ports';
 
-const getPeerRuleTitle = (t, direction: 'ingress' | 'egress', peer: NetworkPolicyPeer) => {
+const getPeerRuleTitle = (
+  t: TFunction,
```

Comment:
Just a piece of information, this isn't related to your changes so I'm not necessarily asking for this change.

We should avoid passing `t` functions into methods. Any method can invoke the static `i18next.t` function safely as the configuration shared between the `t` functions is the same. It would also clean up method signatures from having `t` unnecessarily required.

Reply:
Ok I'll rewrite

Reply:
Hmm, can you elaborate on invoking the static `i18next.t` ? Do you mean by calling `const { t } = useTranslation();` ?

If so, this change is not as straightforward as it seems. When I do so, I get a react warning:

```
Warning: React has detected a change in the order of Hooks called by NetworkPolicyRuleConfigPanel. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useContext                 useContext
2. useState                   useState
3. useRef                     useRef
4. useEffect                  useEffect
5. undefined                  useContext
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

NetworkPolicyRuleConfigPanel@http://localhost:9000/static/create-network-policy-3563d8cb2c6780af5540.js:1752:75
```

It's the first time I see this one, I understand that calls like `useContext` (which is called within `useTranslation`) should not be done conditionally, which is the case here? I guess then I'd need to refactor the component to split in more pieces so that `useTranslation` is called only one time per component.

Reply:
ok, forget what I said, I didn't know we could invoke directly `i18next.t` :-)
Can you explain when it's preferable to use `useTranslation` then?

Reply:
Ah apologizes for the lack of clarity in my original comment. 

- `useTranslation` is a hook based one and thus should be used inside components
- `i18next.t` invoked directly should only be done in utilities (functions that are not components); it pulls from the same config and thus when invoked at runtime will have the correct language setting from the other parts of the UI

A side note, don't use `i18next.t` in the root of any file. The files get parsed once at the first import and thus will lock in the language setting and never change.